### ISSUE TITLE
Terminal window behaves oddly after game ends

### ISSUE THREAD

**Issue by zqqw** [indiscipline/tetronimia#9]
```
~$ tetronimia
|     ##   |
|    ##    |
|    ***   |
|      *   |
|   ****   |
|    **    |
|    **    |
|    ***   |
|     *    |
|    ***   |
|      *   |
|     **   |
|    **    |
|    ***   |
|    *     |
|   ****   |
|    **    |
|    **    |
|    *     |
|   **  *  |
|***** ** *|
____________
Game Over! Final Score: 154, Cleared: 2, Level: 1 
Game settings: "-s=n -r=cw --gameseed vjSdgiHR/AU="
bash: jjjjjjjjjjjjjj: command not found
~$                            ~$ 
^^^^Prompt ends up here
```
Further presses of enter don't do anything, ctrl L clears the screen but it still didn't work properly. Pressing enter did nothing, ls then enter kind of splattered the output across the screen. You can enter commands and they work but you can't see them as they are typed and any resulting onscreen output doesn't appear normally. OS is Artix Linux stable up to date using Bash as the shell, Mate terminal in Mate desktop.

**Comment by indiscipline** [comment]
Never caught this peculiar bug as it's specific to a shell (Bash). Fix on the way.

**Comment by indiscipline** [comment]
Fixed in [v0.2.1](https://github.com/indiscipline/tetronimia/commit/3ed7c9094b1d29fa27cf2c33b0d81dd1ed89b168)

The thread dealing with user input is waiting for it in a blocking call. We currently rely on the OS to clean up after ther program finishes and terminate all the threads. However, when the program quits, the thread gets culled by the OS and the `getch` procedure doesn't get to restore the state of the stdin, so to fix the bug we have to do it manually now. Thanks to @beef331 for the implementation of the exitproc hook which restores the state of the terminal.

This is an insteresting bug, as it was specific to Bash and slipped trough my main testing rigs both on Linux (Fish shell) and Windows (cmd.exe).

Thanks to @zqqw for checking the program and caring enough to report a bug!
